pymc4/gp/mean.py

```diff
@@ -28,76 +49,86 @@ def __mul__(self, mean2):
 
 
 class MeanAdd(Mean):
-    r"""Addition of two or more mean functions
+    r"""
+    Addition of two or more mean functions.
 
     Parameters
     ----------
-    mean1 : {callable, pm.Mean}
+    mean1 : Mean
         First mean function
-    mean2 : {callable, pm.Mean}
+    mean2 : Mean
         Second mean function
     """
 
     def __init__(self, mean1: Mean, mean2: Mean):
         if mean1.feature_ndims != mean2.feature_ndims:
-            raise ValueError("Cannot combine means with different feature_ndims")
+            raise ValueError("Cannot combine means with different feature_ndims.")
         self.mean1 = mean1
         self.mean2 = mean2
 
+    @_inherit_docs(Mean.__call__)
     def __call__(self, X: ArrayLike) -> TfTensor:
         return self.mean1(X) + self.mean2(X)
 
 
 class MeanProd(Mean):
-    r"""Product of two or more mean functions
+    r"""
+    Product of two or more mean functions.
 
     Parameters
     ----------
-    mean1 : {callable, pm.Mean}
+    mean1 : Mean
         First mean function
-    mean2 : {callable, pm.Mean}
+    mean2 : Mean
         Second mean function
     """
 
     def __init__(self, mean1: Mean, mean2: Mean):
         if mean1.feature_ndims != mean2.feature_ndims:
-            raise ValueError("Cannot combine means with different feature_ndims")
+            raise ValueError("Cannot combine means with different feature_ndims.")
         self.mean1 = mean1
         self.mean2 = mean2
 
+    @_inherit_docs(Mean.__call__)
     def __call__(self, X: ArrayLike) -> TfTensor:
         return self.mean1(X) * self.mean2(X)
 
 
 class Zero(Mean):
-    r"""Zero mean
+    r"""
+    Zero mean function.
 
     Parameters
     ----------
     feature_ndims : int, optional
-        number of rightmost dims to include in mean computation
+        number of rightmost dims to include in mean computation. (default=1)
     """
 
+    @_inherit_docs(Mean.__call__)
     def __call__(self, X: ArrayLike) -> TfTensor:
-        X = tf.convert_to_tensor(X)
-        return tf.zeros(X.shape[: -self.feature_ndims])
+        dtyp = dtype_util.common_dtype([X])
```

Comment:
You can call it `dtype`, as its not a reserved word.

Reply:
changed!